app/src/main/java/com/sacramento/movies/ui/movies/MoviesViewModel.kt

```diff
@@ -32,9 +49,48 @@ class MoviesViewModel @Inject constructor(
             pagingSourceFactory = { moviesWithDetailsPagingSource }
         ).flow.cachedIn(viewModelScope)
 
-        lastFetchedMovieResult = newFetchedMovieResult
+        lastFetchedMovieResultFromServer = newFetchedMovieResult
+
+        return newFetchedMovieResult
+    }
+
+    private fun fetchMoviesWithDetailsFromDB() :Flow<PagingData<MovieWithDetailsModel>>{
+        val lastMovieResult = lastFetchedMovieResultFromDB
+        if (lastMovieResult != null) return lastMovieResult
+
+        val newFetchedMovieResult = Pager(config = PagingConfig(
+            pageSize = 100,
+            enablePlaceholders = false,
+            initialLoadSize = 100
+        ),
+            pagingSourceFactory = {moviesWithDetailsPagingSourceDB}
+        ).flow.cachedIn(viewModelScope)
+
+        lastFetchedMovieResultFromServer = newFetchedMovieResult
```

Comment:
lastFetchedMovieResultFromDB should be used here? not lastFetchedMovieResultFromServer